parsers/US_ERCOT.py

```diff
@@ -0,0 +1,379 @@
+)
+RT_PRICES_URL = (
+    f"{US_PROXY}/api/1/services/read/dashboards/systemWidePrices.json?{HOST_PARAMETER}"
+)
+RT_STORAGE_URL = f"{US_PROXY}/api/1/services/read/dashboards/energy-storage-resources.json?{HOST_PARAMETER}"
+
+# These links are found at https://www.ercot.com/gridinfo/generation, and should be updated as new data is released
+HISTORICAL_GENERATION_URL = {
+    "2024": f"{US_PROXY}/files/docs/2024/02/08/IntGenbyFuel2024.xlsx?{HOST_PARAMETER}",
+    "2023": f"{US_PROXY}/files/docs/2023/02/07/IntGenbyFuel2023.xlsx?{HOST_PARAMETER}",
+    "2022": f"{US_PROXY}/files/docs/2022/02/08/IntGenbyFuel2022.xlsx?{HOST_PARAMETER}",
+    "2021": f"{US_PROXY}/files/docs/2021/02/08/IntGenbyFuel2021.xlsx?{HOST_PARAMETER}",
+    "all_previous": f"{US_PROXY}/files/docs/2021/03/10/FuelMixReport_PreviousYears.zip?{HOST_PARAMETER}",
+}
+
+GENERATION_MAPPING = {
+    "Coal and Lignite": "coal",
+    "Hydro": "hydro",
+    "Natural Gas": "gas",
+    "Nuclear": "nuclear",
+    "Other": "unknown",
+    "Power Storage": "battery",
+    "Solar": "solar",
+    "Wind": "wind",
+    "WSL": "battery",
+    "Biomass": "biomass",
+    "Gas": "gas",
+    "Coal": "coal",
+    "Gas-CC": "gas",
+}
+EXCHANGE_MAPPING = {"US-CENT-SWPP": ["dcE", "dcN"], "MX-NE": ["dcL"], "MX-NO": ["dcR"]}
+
+
+def get_data(url: str, session: Session | None = None):
+    """requests ERCOT url and return json"""
+    if not session:
+        session = Session()
+
+    resp: Response = session.get(url, verify=False)
+    response_text = gzip.decompress(resp.content).decode("utf-8")
+    data_json = json.loads(response_text)
+    return data_json
+
+
+def parse_storage_data(session: Session):
+    storage_data_json = get_data(url=RT_STORAGE_URL, session=session)
+
+    storage_by_hour = {}
+    for day_key in ["previousDay", "currentDay"]:
+        if day_key not in storage_data_json:
+            continue
+
+        for entry in storage_data_json[day_key]["data"]:
+            dt = datetime.strptime(entry["timestamp"], "%Y-%m-%d %H:%M:%S%z").replace(
+                tzinfo=TX_TZ
+            )
+            hour_dt = dt.replace(minute=0, second=0, microsecond=0)
+
+            if hour_dt not in storage_by_hour:
+                storage_by_hour[hour_dt] = []
+
+            storage_by_hour[hour_dt].append(entry["netOutput"])
+
+    return storage_by_hour
+
+
+def fetch_live_consumption(
+    zone_key: str,
+    session: Session | None = None,
+    logger: Logger = getLogger(__name__),
+) -> TotalConsumptionList:
+    data_json = get_data(url=RT_CONSUMPTION_URL, session=session)
+    consumption_list = TotalConsumptionList(logger)
+    for key in ["previousDay", "currentDay"]:
+        data_dict = data_json[key]
+        dt = datetime.strptime(data_dict["dayDate"], "%Y-%m-%d %H:%M:%S%z").replace(
+            tzinfo=TX_TZ
+        )
+        for item in data_dict["data"]:
+            if "systemLoad" in item:
+                consumption_list.append(
+                    zoneKey=zone_key,
```

Comment:
Zone key here also need to have the type `ZoneKey` otherwise it causes type errors.